nservicebus/azure-service-bus/understanding-transactions-and-delivery-guarantees.md

```diff
@@ -0,0 +1,46 @@
+---
+title: Understanding internal transactions and delivery guarantees
+reviewed: 2016-04-20
+tags:
+- Azure
+- Cloud
+- Transactions
+---
+
+## Understanding internal transactions and delivery guarantees
+
+### Versions 6 and below
+
+NServiceBus relies on transaction scopes to control how it behaves when it comes to transactions, message dispatching and commit/rollback guarantees. The architecture is schematically represented like this:
+
+![Transactions v6](transactions-v6.png)
+
+The Azure Service Bus library, which is used in the Azure Service Bus transport, has a few requirements, when it is used inside a transaction scope, that must be taken into account.
+* It requires the use of the `Serializable` isolation level when it's used inside a transaction scope. For more information refer to [Transaction Isolation Levels Explained in Details](http://dotnetspeak.com/2013/04/transaction-isolation-levels-explained-in-details) article. Therefore NServiceBus Azure Service Bus transport will always use `Serializable` isolation level to comply with the Azure Service Bus SDK requirements.
+* It also proactively checks to see whether it is used alone inside a transaction and it will throw an exception if used together with another transactional resource such as a database connection. To allow the user code to still use a transaction for it's database connections, in another transaction scope provided by NServiceBus around a handler, the 2 scopes are kept apart and synchronized using an implementation of `IEnlistmentNotification` for both the send and completion operations. (Schematically represented by the orange to yellow star notation in the diagram)
```

Comment:
it's a lot worse, any second resource will make it blow up regardless whether that resource would cause a DTC promotion or not. F.e. it also happens when you send a message to a queue. without this trick nsb would simply not work as you are not able to receive from one queue and send to another in the same scope.